code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/RegularDeploymentUnitInstance.java

```diff
@@ -0,0 +1,173 @@
+                    volumesFromInstancesIds);
+            Object registryCredentialId = launchConfigData.get(ServiceDiscoveryConfigItem.REGISTRYCREDENTIALID
+                    .getRancherName());
+            Long imageId = getImage(String.valueOf(launchConfigData.get(InstanceConstants.FIELD_IMAGE_UUID)),
+                    registryCredentialId != null ? (Integer) registryCredentialId : null);
+            List<Long> networkIds = sdService.getServiceNetworkIds(service);
+
+            Map<Object, Object> properties = new HashMap<Object, Object>();
+            properties.putAll(launchConfigData);
+            properties.put(INSTANCE.ACCOUNT_ID, service.getAccountId());
+            properties.put(INSTANCE.KIND, InstanceConstants.KIND_CONTAINER);
+            properties.put(InstanceConstants.FIELD_NETWORK_IDS, networkIds);
+            properties.put(INSTANCE.IMAGE_ID, imageId);
+            Pair<Instance, ServiceExposeMap> instanceMapPair = exposeMapDao.createServiceInstance(properties,
+                    service,
+                    instanceName);
+            this.instance = instanceMapPair.getLeft();
+            this.exposeMap = instanceMapPair.getRight();
+        }
+        DeferredUtils.nest(new Runnable() {
+            @Override
+            public void run() {
+                objectProcessManager.scheduleStandardProcess(StandardProcess.CREATE, instance, null);
+                objectProcessManager.scheduleStandardProcess(StandardProcess.CREATE, exposeMap, null);
+            }
+        });
+        if (InstanceConstants.STATE_STOPPED.equals(instance.getState())) {
+            DeferredUtils.nest(new Callable<Object>() {
+                @Override
+                public Object call() throws Exception {
+                    objectProcessManager.scheduleProcessInstance(InstanceConstants.PROCESS_START, instance, null);
+                    return null;
+                }
+            });
+        }
+        this.instance = objectManager.reload(this.instance);
+        return this;
+    }
+
+    @Override
+    public DeploymentUnitInstance waitForStart() {
+        this.instance = resourceMonitor.waitFor(objectManager.reload(this.instance), new ResourcePredicate<Instance>() {
+            @Override
+            public boolean evaluate(Instance obj) {
+                return InstanceConstants.STATE_RUNNING.equals(obj.getState());
+            }
+        });
+        return this;
+    }
+
+    @Override
+    public void stop() {
+        objectProcessManager.scheduleProcessInstance(InstanceConstants.PROCESS_STOP, instance, null);
+    }
+
+    @Override
+    public boolean isStarted() {
+        return objectManager.reload(this.instance).getState().equalsIgnoreCase(InstanceConstants.STATE_RUNNING);
+    }
+
+    @Override
+    public boolean needsCleanup() {
+        // TODO Auto-generated method stub
+        return false;
+    }
+
+    protected Long getImage(String imageUuid, Integer registryCredentialId) {
+        Image image;
```

Comment:
@alena1108, @hibooboo2 recently made some changes to the registryCredentialIds.  I think this will cause some problems.  @hibooboo2 Where do you pick the registryCredentialId?  Which class is that?